lib/ansible/modules/files/replace.py

```diff
@@ -224,7 +224,11 @@ def main():
         module.fail_json(rc=257, msg='Path %s does not exist !' % path)
     else:
         f = open(path, 'rb')
-        contents = to_text(f.read(), errors='surrogate_or_strict')
+        data = f.read()
+        try:
```

Comment:
The try: except will lead to tracebacks on python3 as non-utf8 text will be left as bytes and then the byte strings will attempt to be combined with text strings later in the code on python3.